types/choose_sequence/index.js

```diff
@@ -0,0 +1,232 @@
+const getUniqueId = require('../../utils/getUniqueId');
+
+module.exports = ({ card, tags }) => {
+  const cardId = getUniqueId(card.question);
+  const cardGuessesId = `${cardId}guesses`;
+
+  const front = `
+    <div id="${cardId}_wrapper">
+      <div id="question"></div>
+      <div class="questions-wrapper"></div>
+    </div>
+    <button id="${cardId}_checkBtn">Check</button>
+
+    <script>
+      var card = ${JSON.stringify(card)};
+      var emptyGuessMark = '???';
+      var emptyGuessMarkRegexp = /[?]{3}/g;
+
+      if (!window.memoryCards) {
+        window.memoryCards = {};
+      }
+
+      var question = document.querySelector('#question');
+      question.innerHTML = window.memoryCards['${cardGuessesId}'] || card.question.replace(emptyGuessMarkRegexp, getEmptyGuessItem());
+
+      var questionContainer = document.querySelector('.questions-wrapper');
+      var cardAnswers = getCorrectCardAnswers(card.answers);
+      questionContainer.innerHTML = window.memoryCards['${cardId}'] || cardAnswers
+        .sort(function () { return Math.random() - 0.5; })
+        .map(function (el) {
+          return '<button class="answer_options"'
+            + 'data-answer="' + (el.sequenceNumber !== undefined ? el.sequenceNumber : '') + '" '
+            + '>'
+            + el.text
+            + '</button>';
+        })
+        .join('<br />');
+
+      function getCorrectAnswersAmount(question) {
+        return ((question || '').match(emptyGuessMarkRegexp) || []).length;
+      }
+
+      function getEmptyGuessItem() {
+        return '<span class="guess">' + emptyGuessMark + '</span>';
+      }
+
+      function getCorrectCardAnswers(answers) {
+        var correctAnswersAmount = getCorrectAnswersAmount(card.question);
+        for (var i = 0; i < correctAnswersAmount; i++) {
+          answers[i].sequenceNumber = i + 1;
+        }
+        return answers;
+      }
+
+      var guesses = question.getElementsByClassName('guess');
+      var buttons = questionContainer.querySelectorAll('button');
+
+      function showCurrentActiveGuess() {
+        [].forEach.call(guesses, function(el) {
+          el.classList.remove('active-guess');
+        });
+        for (var i = 0; i < guesses.length; i++) {
+          if (guesses[i].innerHTML === emptyGuessMark) {
+            guesses[i].classList.add('active-guess');
+            break;
+          }
+        }
+      }
+
+      function checkAnswer() {
+        var correctAnswers = cardAnswers
+          .filter(function (el) {
+            return el.sequenceNumber !== undefined;
+          })
+          .sort(function (a, b) {
+            return a.sequenceNumber - b.sequenceNumber;
+          });
+        [].forEach.call(guesses, function (el, index) {
+          el.classList.add(el.innerHTML === correctAnswers[index].text ? 'correct-guess' : 'incorrect-guess');
+        });
+
+        [].forEach.call(buttons, function (el) {
```

Comment:
`buttons.map().join()` doesn't work?

Reply:
I tried and faced some small bugs, that's why I've decided to leave as it is.